# Report

In [1]:
from dataset import *

## Data Preprocessing

In [11]:
!python3 dataset.py

loading train...
loading val...
loading test...
Train:	 300
Val:	 150
Test:	 100
Mean: [124.495 118.847  95.293]
Std:  [62.754 59.597 62.425]


### (a)
i). <br />
Mean: [124.495 118.847  95.293]<br />
Std:  [62.754 59.597 62.425]<br /><br />
ii). <br />
Because we are training a model that is representative of the training dataset. This means that the model should be fit based on properties of the training data. We are simply applying the model to validation and testing data; so if we use their properties, it may cause overfitting.

### (b)
![](Data_Preprocessing.png)

## Convolutional Neural Network

### (a)
There are 32*2 = 64 parameters to be learned; they are the weights of the fully connected layer.<br />

### (f)
i). <br />
One reason is that our the model is overfitting the training data so it won't perform well on the validation dataset. Another possible reason is that the the training, validation, and testing dataset aren't distributed similarly. So what the model learns from the training data can't be used to predict the validation data well. <br /><br />

ii). <br />
The model stopped training at epoch 11. With a patience of 10, the model would have to wait until epoch 16 to stop. Based on the training graphs, patience = 5 is a better choice because the graphs are all generally identical. However, patience = 5 takes fewer epoch to reach the performance. A higher patience might be better for training data that has a local minimum; if the patience is high, we won't stop at the local minimum and could potentially get a better model.
![](cnn_training_plot_5.png)
![](cnn_training_plot_10.png)

iii). <br />
